### PR TITLE
Implements AST Customization (#2388)

### PR DESCRIPTION
**Pull request by pca006132**
Tested with `cargo test` in `ast`, `parser` and `compiler` directory (as `cargo test --all` would fail to compile for wasm...)

### CODE REVIEW

**Comment by coolreader18** [comment]
(The clippy error is a separate issue, addressed in #2392)

Is this ready to be merged? Or will you implement custom-type mapping inside rustpython-ast?

File: parser/src/python.lalrpop [hunk 1]
```diff
@@ -69,6 +69,7 @@ PassStatement: ast::Statement = {
     <location:@L> "pass" => {
         ast::Statement {
             location,
+            custom: Default::default(),
```
Comment (coolreader18) [inline parser/src/python.lalrpop:72]: Is there a reason these can't be `custom: ()` as well? If you were looking to eventually have the parser be generic, `custom: ()` can be easily find/replaced eventually.

Comment (pca006132) [reply parser/src/python.lalrpop:72]: I was looking to have the parser be generic later, but `()` seems to be clearer. will fix.

**Comment by pca006132** [comment]
> (The clippy error is a separate issue, addressed in #2392)
> 
> Is this ready to be merged? Or will you implement custom-type mapping inside rustpython-ast?

I think custom-type mapping can be done in a separate PR, as that would probably be pretty large, as the AST involves quite a lot of structs.

**Comment by coolreader18** [comment]
Could you switch out all the `Default::default()`s with `()` ? It's just less noisy, IMO

**Review by coolreader18 — APPROVED** [review]

**Comment by coolreader18** [comment]
Thanks for contributing!

**Pull request merged**